File: mock-auth-server.js
const express = require('express');
const cors = require('cors');
const app = express();

app.use(cors());
app.use(express.json());

// Mock auth endpoint
app.post('/auth/token', (req, res) => {
  const { sub, role } = req.body;
  
  // Create a mock JWT token
  const mockToken = Buffer.from(JSON.stringify({
    sub,
    role,
    exp: Date.now() + 86400000, // 24 hours
    iat: Date.now()
  })).toString('base64');
  
  res.json({
    access_token: `mock.${mockToken}.signature`,
    token_type: 'Bearer',
    expires_in: 86400
  });
});

// Mock health endpoint
app.get('/health', (req, res) => {
  res.json({ status: 'ok' });
});

// Mock workflow endpoints
app.get('/api/v1/workflows/templates', (req, res) => {
  res.json([
    {
      id: 'customer-support',
      name: 'Customer Support Workflow',
      description: 'Handle customer inquiries'
    },
    {
      id: 'knowledge-base',
      name: 'Knowledge Base Workflow',
      description: 'Search and retrieve information'
    }
  ]);
});

const PORT = 8080;
app.listen(PORT, () => {
  console.log(`Mock auth server running on http://localhost:${PORT}`);
});

File: mock_auth_server.py
#!/usr/bin/env python3
from flask import Flask, request, jsonify
from flask_cors import CORS
import base64
import json
import time

app = Flask(__name__)
CORS(app)

@app.route('/auth/token', methods=['POST'])
def auth_token():
    data = request.get_json()
    sub = data.get('sub', 'user')
    role = data.get('role', 'Admin')
    
    # Create mock JWT payload
    payload = {
        'sub': sub,
        'role': role,
        'exp': int(time.time()) + 86400,  # 24 hours
        'iat': int(time.time())
    }
    
    # Create a mock JWT token (not cryptographically secure, just for testing)
    header = base64.b64encode(json.dumps({'alg': 'HS256', 'typ': 'JWT'}).encode()).decode()
    payload_encoded = base64.b64encode(json.dumps(payload).encode()).decode()
    signature = 'mock_signature'
    
    token = f"{header}.{payload_encoded}.{signature}"
    
    return jsonify({
        'access_token': token,
        'token_type': 'Bearer',
        'expires_in': 86400
    })

@app.route('/health', methods=['GET'])
def health():
    return jsonify({'status': 'ok'})

@app.route('/api/v1/workflows/templates', methods=['GET'])
def workflow_templates():
    return jsonify([
        {
            'id': 'customer-support',
            'name': 'Customer Support Workflow',
            'description': 'Handle customer inquiries'
        },
        {
            'id': 'knowledge-base',
            'name': 'Knowledge Base Workflow',
            'description': 'Search and retrieve information'
        }
    ])

if __name__ == '__main__':
    print("Mock auth server running on http://localhost:8080")
    app.run(port=8080, debug=False)